prow/gerrit/adapter/adapter.go

```diff
@@ -65,19 +68,57 @@ type LastSyncTracker interface {
 }
 
 // NewController returns a new gerrit controller client
-func NewController(lastSyncTracker LastSyncTracker, gc gerritClient, prowJobClient prowv1.ProwJobInterface,
-	cfg config.Getter, projectsOptOutHelp map[string][]string) *Controller {
-	projs := map[string]sets.String{}
-	for i, p := range projectsOptOutHelp {
-		projs[i] = sets.NewString(p...)
+func NewController(ctx context.Context, prowJobClient prowv1.ProwJobInterface, op io.Opener,
+	cfg config.Getter, projects, projectsOptOutHelp map[string][]string, cookiefilePath, tokenPathOverride string, lastSyncFallback string) *Controller {
+
+	projectsOptOutHelpMap := map[string]sets.String{}
+	if cfg().Gerrit.OrgReposConfig != nil {
+		projectsOptOutHelpMap = cfg().Gerrit.OrgReposConfig.OptOutHelpRepos()
+	} else {
+		for i, p := range projectsOptOutHelp {
+			projectsOptOutHelpMap[i] = sets.NewString(p...)
+		}
+	}
+	lastSyncTracker := &syncTime{
+		path:   lastSyncFallback,
+		ctx:    ctx,
+		opener: op,
+	}
+	if err := lastSyncTracker.init(projects); err != nil {
+		logrus.WithError(err).Fatal("Error initializing lastSyncFallback.")
 	}
-	return &Controller{
+	gerritClient, err := client.NewClient(projects)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error creating gerrit client.")
+	}
+	gerritClient.Authenticate(cookiefilePath, tokenPathOverride)
+
+	c := &Controller{
 		prowJobClient:      prowJobClient,
 		config:             cfg,
-		gc:                 gc,
+		gc:                 gerritClient,
 		tracker:            lastSyncTracker,
-		projectsOptOutHelp: projs,
+		projectsOptOutHelp: projectsOptOutHelpMap,
 	}
+	go func() {
+		for {
```

Comment:
I think this loop needs some sort of delay. We don't need to check this more than once a second.

Reply:
good point, added a sleep